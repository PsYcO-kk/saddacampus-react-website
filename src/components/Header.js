import React from 'react';
import { Collapse, Navbar, NavbarToggler, NavbarBrand, Nav, NavItem } from 'reactstrap';
import { NavLink } from 'react-router-dom';

export default class Header extends React.Component {
	constructor(){
		super();
		this.state = {
			isOpen: false
		}
	}
	toggle = () => {
		this.setState({
			isOpen: !this.state.isOpen
		});
	}
	render(){
		return (
			<div>
				<Navbar color="light" light expand="lg" className="bg-white border-bottom">
					<div className="container">
						<NavbarBrand href="/">
							<img src="/assets/images/brand/16-9.png" width="50" className="d-inline-block align-top" alt="" />
							&nbsp; Saddacampus
						</NavbarBrand>
						<NavbarToggler onClick={this.toggle} />
						<Collapse isOpen={this.state.isOpen} navbar>
							<Nav className="ml-auto" navbar>
								<NavItem>
									<NavLink to="/about-us" activeClassName="active" className="nav-link">About Us</NavLink>
								</NavItem>
								<NavItem>
									<NavLink to="/privacy-policy" activeClassName="active" className="nav-link">Privacy Policy</NavLink>
								</NavItem>
							</Nav>
						</Collapse>
					</div>
				</Navbar>
			</div>
		);
	}
}
