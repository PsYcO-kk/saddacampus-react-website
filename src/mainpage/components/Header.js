import React from 'react';
import { Collapse, Navbar, NavbarToggler, NavbarBrand, Nav, NavItem ,  ButtonDropdown, DropdownToggle, DropdownMenu, DropdownItem , Button } from 'reactstrap';
import { NavLink } from 'react-router-dom';
import FontAwesomeIcon from '@fortawesome/react-fontawesome'
import {faCog} from "@fortawesome/fontawesome-free-solid"

const style = {
    "width" : "10%",
}
const paddingTop = {
	paddingTop : "15px"
}
const NavItemPadding = {
	"paddingRight"  : "40px"
}
const NavFixed = {
	"width" : "100%",
	"position" : "fixed",
	"zIndex" : "99"
}
var styleSetting = {
    "backgroundColor" : "white" ,
    "border" : "none",
    "color": "black",
	"boxShadow"  :"none",
	"marginTop" : "8px"
}
var positionSettings = {
	"position":"relative",
	"bottom":"4px"
}
var dropdownBox = {
	"borderRadius"  : "0px"
}
export default class Header extends React.Component {
	constructor(){
		super();
		this.state = { isOpen: false , isOpen2 : false };
	}
	toggle = () => {
		this.setState({
			isOpen: !this.state.isOpen
		});
	}
	toggle_ = () => {
		this.setState({
			isOpen2 : !this.state.isOpen2
		});
	}
	render(){
		return (
			<div style={NavFixed} >  
				<Navbar color="light" light expand="lg" className="bg-white" style={paddingTop} >
					<div className="container">
						<NavbarBrand href="/" className="logoName" >
							<img src="/assets/images/brand/16-9.png" style={style} className="d-inline-block align-top" alt="" />
							&nbsp; Food delivery
						</NavbarBrand>
						<NavbarToggler onClick={this.toggle} />
						<Collapse isOpen={this.state.isOpen} navbar>
							<Nav className="ml-auto" navbar>
								<NavItem style={NavItemPadding}>
                        		<NavLink to="/profile" className="nav-link">Profile</NavLink>
								</NavItem>
								<NavItem  style={positionSettings} >
									<ButtonDropdown isOpen={this.state.isOpen2} toggle={this.toggle_} className="buttongroup">
									<DropdownToggle style={styleSetting} >
									<img src="/assets/images/icon_setting.png" className="settingIcon" />
									</DropdownToggle>
									<DropdownMenu style={dropdownBox} >
										<DropdownItem >Edit Profile</DropdownItem>
										<DropdownItem >My Orders</DropdownItem>
										<DropdownItem divider className="divider"  ></DropdownItem>
										<DropdownItem>Contact Us</DropdownItem>
										<DropdownItem>Privacy Policy</DropdownItem>
										<DropdownItem divider className="divider"  ></DropdownItem>
										<DropdownItem>Log out</DropdownItem>
									</DropdownMenu>
									</ButtonDropdown>
								</NavItem>
							</Nav>
						</Collapse>
					</div>
				</Navbar>
			</div>
		);
	}
}
