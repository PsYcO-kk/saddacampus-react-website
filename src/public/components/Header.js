import React from 'react';
import { Container, Navbar, NavbarBrand, Nav, NavItem } from 'reactstrap';
import { NavLink } from 'react-router-dom';

export default class Header extends React.Component {
	constructor(){
		super();
		this.state = { dropdownOpen: false };
	}
	componentDidMount(){
		window.addEventListener("click", (event) => {
			if(!event.target.matches('#profile-options')){
				this.setState({
					dropdownOpen: false
				});
			}
		});
	}
	toggle = () => {
		this.setState({
			dropdownOpen: !this.state.dropdownOpen
		});
	}
	componentWillUnmount() {
		window.removeEventListener("click", this.updateDimensions);
	}
	render(){
		return (
			<Navbar color="light" light expand="lg" sticky="top" className="bg-white border-bottom" style={{ boxShadow: '0px 8px 6px -7px #ccc' }}>
				<Container>
					<NavbarBrand href="/" style={{ fontWeight: 'bold' }}>
						<img src="/assets/images/brand/16-9.png" width="75" className="d-inline-block" alt="" />
						 &nbsp;&nbsp; SADDACAMPUS
					</NavbarBrand>
					<Nav className="ml-auto" navbar style={{ flexDirection: 'row' }}>
						<NavItem>
							<NavLink to="/" className="nav-link">
								<img className="img-fluid" width="30" src="assets/images/svg/home.svg" />
							</NavLink>
						</NavItem>
						&nbsp;&nbsp;
						<NavItem>
							<NavLink to="/help" className="nav-link">
								<img className="img-fluid" width="30" src="assets/images/svg/questions-circular-button.svg" />
							</NavLink>
						</NavItem>
						&nbsp;&nbsp;
						<NavItem>
							<div className="nav-link" style={{ position: 'relative' }}>
								<img id="profile-options" className="img-fluid" width="30" src="assets/images/svg/user.svg" onClick={this.toggle} />
								<div style={{ position: 'absolute', top: '100%', display: (this.state.dropdownOpen ? 'block' : 'none'), minWidth: '150px', overflow: 'auto', zIndex: 1, backgroundColor: 'rgba(255,255,255,0.9)', boxShadow: '0 2px 6px 0 #ccc' }}>
									<div style={{ padding: '5px 10px' }}>Header</div>
									<div style={{ padding: '5px 10px' }}>Header</div>
									<div style={{ padding: '5px 10px' }}>Header</div>
								</div>
							</div>
						</NavItem>
					</Nav>
				</Container>
			</Navbar>
		);
	}
}
