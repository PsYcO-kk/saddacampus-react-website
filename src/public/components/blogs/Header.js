import React from 'react';
import { Col, Collapse, Container, Navbar, NavbarToggler, NavbarBrand, Nav, NavItem, Row } from 'reactstrap';
import { NavLink } from 'react-router-dom';
import Slider from "react-slick";

import "slick-carousel/slick/slick.css";
import "slick-carousel/slick/slick-theme.css";

export default class Header extends React.Component {
	constructor(){
		super();
		this.state = { isOpen: false };
	}
	toggle = () => {
		this.setState({
			isOpen: !this.state.isOpen
		});
	}
	render(){
		return (
			<Navbar color="light" light expand="lg" sticky="top" className="bg-white border-bottom">
				<Container>
					<NavbarBrand href="/">
						<img src="/assets/images/brand/16-9.png" width="100" className="d-inline-block align-top" alt="" />
					</NavbarBrand>
					<NavbarToggler onClick={this.toggle} />
					<Collapse isOpen={this.state.isOpen} navbar>
						<Nav className="ml-auto" navbar>
							<NavItem>
								<NavLink to="/home" activeClassName="active" className="nav-link" style={{ 'fontWeight': 'bold' }}>Home</NavLink>
							</NavItem>
							<NavItem>
								<NavLink to="/food-delivery" activeClassName="active" className="nav-link" style={{ 'fontWeight': 'bold' }}>Food Delivery</NavLink>
							</NavItem>
							<NavItem>
								<NavLink to="/video-lectures" activeClassName="active" className="nav-link" style={{ 'fontWeight': 'bold' }}>Video Lectures</NavLink>
							</NavItem>
							<NavItem>
								<NavLink to="/internships" activeClassName="active" className="nav-link" style={{ 'fontWeight': 'bold' }}>Internships</NavLink>
							</NavItem>
							<NavItem>
								<NavLink to="/blogs" activeClassName="active" className="nav-link" style={{ 'fontWeight': 'bold' }}>Blogs</NavLink>
							</NavItem>
							<NavItem>
								<NavLink to="/help" activeClassName="active" className="nav-link" style={{ 'fontWeight': 'bold' }}>Help</NavLink>
							</NavItem>
						</Nav>
					</Collapse>
				</Container>
			</Navbar>
		);
	}
}
