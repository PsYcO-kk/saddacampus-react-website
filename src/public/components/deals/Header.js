import React from 'react';
import { Collapse, Container, Navbar, NavbarToggler, NavbarBrand, Nav, NavItem } from 'reactstrap';
import { NavLink } from 'react-router-dom';
import Slider from "react-slick";

import "slick-carousel/slick/slick.css";
import "slick-carousel/slick/slick-theme.css";

export default class Header extends React.Component {
	constructor(){
		super();
		this.state = { isOpen: false };
	}
	toggle = () => {
		this.setState({
			isOpen: !this.state.isOpen
		});
	}
	render(){
		const settings = {
			infinite: false,
			speed: 500,
			slidesToScroll: 1,
			slidesToShow: 4,
			swipeToSlide: true,
			responsive: [
				{
					breakpoint: 1024,
					settings: {
						slidesToShow: 3,
						slidesToScroll: 1,
					}
				},
				{
					breakpoint: 600,
					settings: {
						slidesToShow: 2,
						slidesToScroll: 1,
					}
				},
				{
					breakpoint: 480,
					settings: {
						slidesToShow: 1,
						slidesToScroll: 1
					}
				}
			]
		};
		return (
			<div>
				<Navbar color="light" light expand="lg" className="bg-white border-bottom">
					<Container>
						<NavbarBrand href="/">
							<img src="/assets/images/brand/16-9.png" width="100" className="d-inline-block align-top" alt="" />
						</NavbarBrand>
						<NavbarToggler onClick={this.toggle} />
						<Collapse isOpen={this.state.isOpen} navbar>
							<Nav className="ml-auto" navbar>
								<NavItem>
									<a href="https://play.google.com/store/apps/details?id=com.saddacampus.app" className="nav-link" style={{ 'fontWeight': 'bold' }}>Get the App</a>
								</NavItem>
								<NavItem>
									<NavLink to="/about-us" activeClassName="active" className="nav-link" style={{ 'fontWeight': 'bold' }}>About Us</NavLink>
								</NavItem>
								<NavItem>
									<NavLink to="/privacy-policy" activeClassName="active" className="nav-link" style={{ 'fontWeight': 'bold' }}>Privacy Policy</NavLink>
								</NavItem>
							</Nav>
						</Collapse>
					</Container>
				</Navbar>
				<div className="deals-header">
					<Container>
						<Slider {...settings}>
							<div>
								<div className="thumb">
									<img className="img-fluid" src="assets/images/bright-color-eyes-413998.jpg" />
								</div>
							</div>
							<div>
								<div className="thumb">
									<img className="img-fluid" src="assets/images/bright-color-eyes-413998.jpg" />
								</div>
							</div>
							<div>
								<div className="thumb">
									<img className="img-fluid" src="assets/images/bright-color-eyes-413998.jpg" />
								</div>
							</div>
							<div>
								<div className="thumb">
									<img className="img-fluid" src="assets/images/bright-color-eyes-413998.jpg" />
								</div>
							</div>
							<div>
								<div className="thumb">
									<img className="img-fluid" src="assets/images/bright-color-eyes-413998.jpg" />
								</div>
							</div>
							<div>
								<div className="thumb">
									<img className="img-fluid" src="assets/images/bright-color-eyes-413998.jpg" />
								</div>
							</div>
							<div>
								<div className="thumb">
									<img className="img-fluid" src="assets/images/bright-color-eyes-413998.jpg" />
								</div>
							</div>
							<div>
								<div className="thumb">
									<img className="img-fluid" src="assets/images/bright-color-eyes-413998.jpg" />
								</div>
							</div>
						</Slider>
						<div className="row section-title">
							<h1>Deals from all around</h1>
							<p>There are always some things you need to know and once in college you need all the help available.</p>
						</div>
					</Container>
				</div>
			</div>
		);
	}
}
