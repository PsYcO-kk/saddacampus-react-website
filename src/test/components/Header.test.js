import {shallow} from 'enzyme';
import React from 'react';
import Header from './../../components/Header.js'

test("testing for Header page component" , ()=>{
    const wrapper = shallow(<Header />)
    expect(wrapper).toMatchSnapshot();
});

test("testing the click function of Navbar Toggle" , ()=>{
    const wrapper = shallow(<Header />);
    const isOpen = wrapper.state("isOpen");
    wrapper.find('NavbarToggler').simulate('click');
    expect(wrapper.state("isOpen")).toBe(!isOpen);
})