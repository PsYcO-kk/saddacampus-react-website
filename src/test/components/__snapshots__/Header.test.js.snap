// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing for Header page component 1`] = `
<div>
  <Navbar
    className="bg-white border-bottom"
    color="light"
    expand="lg"
    light={true}
    tag="nav"
  >
    <div
      className="container"
    >
      <NavbarBrand
        href="/"
        tag="a"
      >
        <img
          alt=""
          className="d-inline-block align-top"
          src="/assets/images/brand/16-9.png"
          width="50"
        />
        Â  Saddacampus
      </NavbarBrand>
      <NavbarToggler
        onClick={[Function]}
        tag="button"
        type="button"
      />
      <Collapse
        appear={false}
        enter={true}
        exit={true}
        in={false}
        isOpen={false}
        mountOnEnter={false}
        navbar={true}
        onEnter={[Function]}
        onEntered={[Function]}
        onEntering={[Function]}
        onExit={[Function]}
        onExited={[Function]}
        onExiting={[Function]}
        tag="div"
        timeout={350}
        unmountOnExit={false}
      >
        <Nav
          className="ml-auto"
          navbar={true}
          tag="ul"
          vertical={false}
        >
          <NavItem
            tag="li"
          >
            <NavLink
              activeClassName="active"
              aria-current="page"
              className="nav-link"
              to="/about-us"
            >
              About Us
            </NavLink>
          </NavItem>
          <NavItem
            tag="li"
          >
            <NavLink
              activeClassName="active"
              aria-current="page"
              className="nav-link"
              to="/privacy-policy"
            >
              Privacy Policy
            </NavLink>
          </NavItem>
        </Nav>
      </Collapse>
    </div>
  </Navbar>
</div>
`;
